src/classes/Consumer.js

```diff
@@ -34,28 +44,78 @@ class Consumer {
     const offer = await this.connection.createOffer();
     await this.connection.setLocalDescription(offer);
     console.log(this.clientId);
-    this.socket.emit('consumerHandshake', {
-      description: this.connection.localDescription,
-      clientId: this.clientId,
-      remotePeerId: this.remotePeerId,
-    });
+
+    const payload = {
+      action: 'handshake',
+      data: {
+        type: 'consumer',
+        sender: this.sfuId,
+        receiver: this.clientId,
+        remotePeerId: this.remotePeerId,
+        description: this.connection.localDescription,
+      },
+    };
+
+    console.log('Sending consumer offer');
+    console.log(payload);
+
+    this.socket.send(JSON.stringify(payload));
   }
 
   addTrack(track) {
     this.connection.addTrack(track);
   }
 
-  async handshake(description, candidate) {
+  modifyIceAttributes(sdp) {
+    const iceAttributesRegex = /a=(ice-pwd:|ice-ufrag:)(.*)/gi;
+    const modifiedSdp = sdp.replace(
+      iceAttributesRegex,
+      (match, attribute, value) => {
+        // Replace spaces with '+'
+        const modifiedValue = value.replace(/ /g, '+');
```

Comment:
can you include a comment here that outlines why this has to be done in case someone works on it who is unaware of the problem?

Reply:
Yeah and this is only a temporary fix and I think one of the reasons some of the ICE Candidates fail, because I'm only checking two properties of the SDP here. The fix SHOULD be Stringifying the Description/Candidate and then using EncodeURIComponent to essentially escape all of those characters so they aren't being replaced when being passed to the SQS